### DISCUSSION TITLE
C# compiler warning on JobWasExecuted override

### DISCUSSION THREAD

**Discussion by glenserraview** [quartznet/quartznet#1516] [Q&A] [answered]
I have a project using quartz.net. I'm building in Visual Studio 2019 and the target framework is .NET core 3.1, so I believe it will be using C# 8.0

I've overridden JobWasExecuted

```
public override async Task JobWasExecuted(
        IJobExecutionContext context, 
        JobExecutionException? jobException, 
        CancellationToken cancellationToken = default)
```

And when I compile I get the following warning:

> Warning CS8632 The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.

This is caused by the JobExecutionException being nullable.

Our workaround is to add #nullable enable which removes the warning, but it also doesn't allow the additional compile protection.

Ideally the JobExecuctionException parameter would not be nullable, which would remove the need for this workaround.

**Comment by lahma** [comment, marked as answer]
Why don't you just remove the `?` from the signature if you aren't using nullable reference types?

**Reply by glenserraview** [reply]
It seems that this is the right approach. I thought I needed to mimic the interface definition. Thanks and sorry for the dumb question.
